packages/@aws-cdk/aws-stepfunctions-tasks/lib/ecs/run-task.ts

```diff
@@ -0,0 +1,363 @@
+}
+
+/**
+ * Configuration options for the ECS launch type
+ */
+export interface EcsLaunchTargetConfig {
+  /**
+   * Additional parameters to pass to the base task
+   *
+   * @default - No additional parameters passed
+   */
+  readonly parameters?: { [key: string]: any };
+}
+
+/**
+ * Properties to define an ECS service
+ */
+export interface EcsFargateLaunchTargetOptions {
+  /**
+   * Fargate platform version to run this service on
+   *
+   * Unless you have specific compatibility requirements, you don't need to
+   * specify this.
+   *
+   * @default FargatePlatformVersion.LATEST
+   */
+  readonly platformVersion?: ecs.FargatePlatformVersion;
+}
+
+/**
+ * Options to run an ECS task on EC2 in StepFunctions and ECS
+ */
+export interface EcsEc2LaunchTargetOptions {
+  /**
+   * Placement constraints
+   *
+   * @default  - None
+   */
+  readonly placementConstraints?: ecs.PlacementConstraint[];
+
+  /**
+   * Placement strategies
+   *
+   * @default - None
+   */
+  readonly placementStrategies?: ecs.PlacementStrategy[];
+}
+
+/**
+ * Configuration for running an ECS task on Fargate
+ *
+ * @see https://docs.aws.amazon.com/AmazonECS/latest/userguide/launch_types.html#launch-type-fargate
+ */
+export class EcsFargateLaunchTarget implements IEcsLaunchTarget {
+  constructor(private readonly options?: EcsFargateLaunchTargetOptions) {}
+
+  /**
+   * Called when the Fargate launch type configured on RunTask
+   */
+  public bind(_task: EcsRunTask, taskDefinition: ecs.ITaskDefinition, _cluster?: ecs.ICluster): EcsLaunchTargetConfig {
+    if (!taskDefinition.isFargateCompatible) {
+      throw new Error('Supplied TaskDefinition is not compatible with Fargate');
+    }
+
+    return {
+      parameters: {
+        LaunchType: 'FARGATE',
+        PlatformVersion: this.options?.platformVersion,
+      },
+    };
+  }
+}
+
+/**
+ * Configuration for running an ECS task on EC2
+ *
+ * @see https://docs.aws.amazon.com/AmazonECS/latest/userguide/launch_types.html#launch-type-ec2
+ */
+export class EcsEc2LaunchTarget implements IEcsLaunchTarget {
+  constructor(private readonly options?: EcsEc2LaunchTargetOptions) {}
+  /**
+   * Called when the EC2 launch type is configured on RunTask
+   */
+  public bind(_task: EcsRunTask, taskDefinition: ecs.ITaskDefinition, cluster?: ecs.ICluster): EcsLaunchTargetConfig {
+    if (!taskDefinition.isEc2Compatible) {
+      throw new Error('Supplied TaskDefinition is not compatible with Fargate');
+    }
+
+    if (!cluster?.hasEc2Capacity) {
+      throw new Error('Cluster for this service needs Ec2 capacity. Call addXxxCapacity() on the cluster.');
+    }
+
+    return {
+      parameters: {
+        LaunchType: 'EC2',
+        PlacementConstraints: noEmpty(flatten((this.options?.placementConstraints ?? []).map((c) => c.toJson().map(uppercaseKeys)))),
+        PlacementStrategy: noEmpty(flatten((this.options?.placementStrategies ?? []).map((c) => c.toJson().map(uppercaseKeys)))),
```

Comment:
Add a code comment here on what's going on with an example, to help the reader. Provide one example input for placementConstraints and an example output.

Reply:
Also, should all this transformation be a method on `PlacementConstraints` and `PlacementStrategy`, or is this specific to this SFN task?

(no need to change it here but curious)

Reply:
Ping

Reply:
missed this comment, thanks for the ping. sure I'll add a comment with an example

Reply:
> Also, should all this transformation be a method on `PlacementConstraints` and `PlacementStrategy`, or is this specific to this SFN task?
> 
> (no need to change it here but curious)

it currently is, but it's internal to that module as we wouldn't want to export it either. This transformation (of flattening and upper casing) seems to be done in a few places in our repo. I'm not addressing it in this PR, but there's probably a cleanup activity to follow up on